src/taco-remote/lib/buildManager.ts

```diff
@@ -369,10 +369,20 @@ class BuildManager {
         if (fs.existsSync(changeListFile)) {
             buildInfo.changeList = JSON.parse(fs.readFileSync(changeListFile, { encoding: "utf-8" }));
             if (buildInfo.changeList) {
+                buildInfo.changeList.deletedFiles = buildInfo.changeList.deletedFiles.map(function (deletedFile: string): string {
+                    // Convert all \ and / characters in the path string into platform-appropriate path separators
+                    return deletedFile.split("\\").join(path.sep).split("/").join(path.sep);
+                });
                 buildInfo.changeList.deletedFiles.forEach(function (deletedFile: string): void {
-                    if (!deletedFile.match(/^plugins[\/]/)) {
+                    
+                    if (deletedFile.split(path.sep)[0] === "plugins") {
```

Comment:
Clients should not be sending sending paths like foo/../bar/../baz, so if they do so we will try not to do anything bad, but I don't think it is necessary to be perfect there. If we do resolve, we still might end up with something like "../cordovaApp/plugins" which (in the context of where it is eventually used) is equivalent to "plugins" but that's harder to catch, so we'd need to do something like resolve it to an absolute location, then re-relativize it, and then see if it started with plugins.